ruby/ql/lib/codeql/ruby/frameworks/Ldap.qll

```diff
@@ -0,0 +1,89 @@
+/**
+ * Provides modeling for `net-ldap` a ruby library for LDAP.
+ */
+
+private import ruby
+private import codeql.ruby.ApiGraphs
+private import codeql.ruby.dataflow.FlowSummary
+private import codeql.ruby.Concepts
+
+/**
+ * Provides modeling for `net-ldap` a ruby library for LDAP.
+ */
+module NetLdap {
+  /**
+   * Flow summary for `Net::LDAP.new`. This method establishes a connection to a LDAP server.
+   */
+  private class LdapSummary extends SummarizedCallable {
+    LdapSummary() { this = "Net::LDAP.new" }
+
+    override MethodCall getACall() { result = any(NetLdapConnection l).asExpr().getExpr() }
+
+    override predicate propagatesFlowExt(string input, string output, boolean preservesValue) {
+      input = "Argument[0]" and output = "ReturnValue" and preservesValue = false
+    }
+  }
+
+  /** Net LDAP Api Node */
+  private API::Node ldap() { result = API::getTopLevelMember("Net").getMember("LDAP") }
+
+  /** A call that establishes a LDAP Connection */
+  private class NetLdapConnection extends DataFlow::CallNode {
+    NetLdapConnection() { this in [ldap().getAnInstantiation(), ldap().getAMethodCall(["open"])] }
+
+    predicate useSsl() {
+      this.getKeywordArgument("encryption").getConstantValue().isStringlikeValue("simple_tls")
+      or
+      this.getAMethodCall("encryption")
+          .getArgument(0)
+          .getConstantValue()
+          .isStringlikeValue(":simple_tls")
+      or
+      this.getAMethodCall("encryption")
+          .getKeywordArgument("method")
+          .getConstantValue()
+          .isStringlikeValue("simple_tls")
+    }
+
+    DataFlow::Node getAuthValue(string arg) {
+      result = this.getKeywordArgument("auth").(DataFlow::CallNode).getKeywordArgument(arg)
```

Comment:
I initially found the cast to `DataFlow::CallNode` a bit confusing - I think this covers cases like `Net::LDAP.new auth: { pass: 'some password' }` where the `"auth"` argument is a hash literal? I'd suggest maybe:
```suggestion
      result =
        this.getKeywordArgument("auth")
            .(DataFlow::HashLiteralNode)
            .getElementFromKey(any(Ast::ConstantValue cv | cv.isStringlikeValue(arg)))
    }
```

The `ConstantValue` wrangling is a bit messy, but I the cast to `HashLiteralNode` a bit easier to understand overall.